### PR TITLE
Joshua s brown/fix python cpplint to call python3

### LINKED ISSUE #398: Detect absence of `python`
I have `python2` and `python3` but not `python`, which results in a lot of build errors. We should deal with this. At least warn if there is no `python` and better yet detect what versions are available and choose one that exists.

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Fixes the issue raised by @jlippuner about the cpp linter. In this pr cmake will warn if python3 is not found it also specifically calls python3 in the Lint.cmake file. 

## PR Checklist

- [x] Code passes cpplint
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
It looks like if a python3 interpreter isn't found, the lint target is just not build but the build process doesn't die? If so, that's great.

File: cmake/Lint.cmake [hunk 1]
```diff
@@ -52,15 +52,22 @@ function(lint_target TARGET_NAME)
     get_target_property(TARGET_SOURCES ${TARGET_NAME} SOURCES)
     get_target_property(TARGET_SOURCE_DIR ${TARGET_NAME} SOURCE_DIR)
 
-    set(TARGET_OUTPUTS)
-    foreach(SOURCE ${TARGET_SOURCES})
-        lint_file(${TARGET_SOURCE_DIR} ${SOURCE} ${SOURCE}.lint)
-        list(APPEND TARGET_OUTPUTS ${SOURCE}.lint)
-    endforeach()
+    if( NOT Python3_Interpreter_FOUND)
+      find_package(Python3 REQUIRED COMPONENTS Interpreter)
```
Comment (AndrewGaspar) [inline cmake/Lint.cmake]: `REQUIRED` will result in a `FATAL_ERROR` if Python3 isn't found, which doesn't seem to be your intention based on the check after this. You probably want to remove `REQUIRED`.

**Review by AndrewGaspar — CHANGES_REQUESTED** [review]

File: cmake/Lint.cmake (hunk 1 above)
Comment (JoshuaSBrown) [reply cmake/Lint.cmake]: Good point!

**Review by AndrewGaspar — APPROVED** [review]

**Review by jlippuner — APPROVED** [review]
Thanks @JoshuaSBrown! I tested this on my system without `python` and it works (the master branch fails, so this fixes the issue).

**Pull request merged**